core/src/VolatilityOracle.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import {IUniswapV3Pool} from "v3-core/contracts/interfaces/IUniswapV3Pool.sol";
+
+import {MAX_SIGMA, IV_SCALE, IV_CHANGE_PER_SECOND, ORACLE_LOOKBACK, FEE_GROWTH_GLOBALS_SAMPLE_PERIOD} from "./libraries/constants/Constants.sol";
+import {Oracle} from "./libraries/Oracle.sol";
+import {Volatility} from "./libraries/Volatility.sol";
+
+/// @title VolatilityOracle
+/// @author Aloe Labs, Inc.
+/// @dev "Test everything; hold fast what is good." - 1 Thessalonians 5:21
+contract VolatilityOracle {
+    struct LastWrite {
+        uint8 index;
+        uint32 time;
+        uint216 iv;
+    }
+
+    mapping(IUniswapV3Pool => Volatility.PoolMetadata) public cachedMetadata;
+
+    mapping(IUniswapV3Pool => Volatility.FeeGrowthGlobals[60]) public feeGrowthGlobals;
+
+    mapping(IUniswapV3Pool => LastWrite) public lastWrites;
+
+    function prepare(IUniswapV3Pool pool) external {
+        cachedMetadata[pool] = _getPoolMetadata(pool);
+
+        if (lastWrites[pool].time == 0) {
+            feeGrowthGlobals[pool][0] = _getFeeGrowthGlobalsNow(pool);
+            lastWrites[pool] = LastWrite({index: 0, time: uint32(block.timestamp), iv: uint216(MAX_SIGMA)});
```

Comment:
Initially, we assume the IV is `MAX_SIGMA` - why? How did we determine this value again?

Reply:
To be extra safe, `Borrower` clamps values received from VolatilityOracle to be between `MIN_SIGMA` and `MAX_SIGMA`. When the VolatilityOracle is first initialized, it's better to be on the upper end of that (more conservative) than the lower end (more risky). But we don't want to set it at some absurdly high value like 2e18 because of the constraint on the derivative (`IV_CHANGE_PER_SECOND`)